defaultNatsServiceWithTrace.go

```diff
@@ -0,0 +1,102 @@
+package natsio
+
+import (
+	"bytes"
+	"context"
+	"github.com/nats-io/nats.go"
+	"github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+	"github.com/orchestd/dependencybundler/interfaces/configuration"
+	"github.com/orchestd/dependencybundler/interfaces/log"
+	. "github.com/orchestd/servicereply"
+	"go.uber.org/fx"
+	"time"
+)
+
+func NewDefaultNatsServiceWithTrace(lc fx.Lifecycle, tracer opentracing.Tracer, config configuration.Config, logger log.Logger) NatsService {
+	connect = func(natsUrl string, options ...nats.Option) (NatsConnection, error) {
+		con, err := nats.Connect(natsUrl, options...)
+		if err != nil {
+			return nil, err
+		}
+		return natsServiceWithTrace{
+			tracer: tracer,
+			logger: logger,
+			nc:     con,
+		}, nil
+	}
+
+	return NewDefaultNatsServiceWithJWTAuth(lc, config, logger)
+}
+
+type natsServiceWithTrace struct {
+	tracer opentracing.Tracer
+	logger log.Logger
+	nc     NatsConnection
+}
+
+func (n natsServiceWithTrace) Close() {
+	n.nc.Close()
+}
+
+func (n natsServiceWithTrace) Publish(subj string, msg []byte) error {
```

Comment:
can we create a nats connection with NO trace
I'm thinking tracker messages (driver location changed) may not need logging in jaeger